protocol/rpcconsumer/rpcconsumer.go

```diff
@@ -97,12 +97,14 @@ type RPCConsumer struct {
 }
 
 // spawns a new RPCConsumer server with all it's processes and internals ready for communications
-func (rpcc *RPCConsumer) Start(ctx context.Context, txFactory tx.Factory, clientCtx client.Context, rpcEndpoints []*lavasession.RPCEndpoint, requiredResponses int, cache *performance.Cache, strategy provideroptimizer.Strategy, metricsListenAddress string, maxConcurrentProviders uint) (err error) {
+func (rpcc *RPCConsumer) Start(ctx context.Context, txFactory tx.Factory, clientCtx client.Context, rpcEndpoints []*lavasession.RPCEndpoint, requiredResponses int, cache *performance.Cache, strategy provideroptimizer.Strategy, metricsListenAddress string, maxConcurrentProviders uint, usageServerAddr string) (err error) {
```

Comment:
i dont like the fact that we just propagate these type of info. next time we need another address we will also propagate it? seems like its time to move to a struct with all the fields so we propagate this struct instead both merics and relay analytics (by the way lets agree on one name and stick with it ? |

"usageServerAddr", in most places you called it relay server here its usage server.

I think a proper name would be "analytics server" / "usage analytic server"

Reply:
i would first delete some other code - and then make the struct - there are things you can delete after this new relay server works

Reply:
there is some code we can delete before - all the old relay server
then it will be easier to see which struct will be best